File: sanidad-backend/src/main/java/com/farmacia/sanidadbackend/dto/DevolucionProveedorRequest.java
package com.farmacia.sanidadbackend.dto;

import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.util.List;

@Data
public class DevolucionProveedorRequest {

    @NotNull
    private Long loteId;

    @NotNull
    private Long solicitadoPorId;

    private String motivo; // opcional

    @NotEmpty
    private List<DevolucionProveedorDetalleRequest> detalles;
}

File: SanidadMovil/app/src/main/java/com/sanidad/movil/data/remote/dto/DevolucionProveedorRequest.kt
package com.sanidad.movil.data.remote.dto

data class DevolucionProveedorRequest(
    val loteId: Long,
    val solicitadoPorId: Long,
    val motivo: String? = null,
    val detalles: List<DevolucionProveedorDetalleRequest>
)